internal/common/actions/create_source_dir/create_source_dir.js

```diff
@@ -0,0 +1,181 @@
+const fs = require("fs-extra");
+const path = require("path");
+const { BazelAction, ensureArray } = require("../run_js/BazelAction");
+
+/**
+ * Action for creating a directory with the passed in files for symlinking and
+ * copying.
+ *
+ * The files can follow a uri syntax with params:
+ * recurse=false (default)
+ * symlink=true (default)
+ *
+ *
+ * The files to be populated have flags set on them for the appropriate actions:
+ * [flags]:./file/path
+ * Possible flags:
+ * s: Symlink
+ *   s:./some/dir/
+ * c: Copy
+ *   c:/some/file
+ * r: Recurse
+ *   rs:./some/dir/
+ * m: Module (put it in node_modules)
+ * mr: Recursive module (add all the modules to node_modules)
```

Comment:
What does "all the modules" mean here? This module and all its dependencies?

Reply:
This should probably be `mrs:./some/node_modules` basically it is able to take in a node modules directory and do a 1 deep symlinking of it. It will go an additional layer down for directories that begin with `"@"`.

Reply:
OK, do you mind updating the code comment with this info?